internal/handlers/common/aggregations/operators/operators.go

```diff
@@ -67,12 +50,17 @@ type Operator interface {
 func NewOperator(doc any) (Operator, error) {
 	operatorDoc, ok := doc.(*types.Document)
 	if !ok {
-		// TODO: https://github.com/FerretDB/FerretDB/pull/2789
-		return nil, ErrWrongType
+		return nil, newOperatorError(
+			ErrWrongType,
+			"Invalid type of operator field (expected document)",
+		)
```

Comment:
Could you add test for this please :pray:

Reply:
As I wanted to move most of the parsing logic to the `NewOperator`, this line is unreachable for now, as `NewOperator` is called from type switches that already check for `*types.Document`.